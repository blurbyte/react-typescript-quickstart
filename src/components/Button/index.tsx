import styled from '@styles/styledComponents';
import { Colors } from '@styles/colors';

const Button = styled.button`
  background: rgba(40, 40, 40, 0.05);
  border: none;
  color: ${Colors.blue};
  border: 1px solid transparent;
  font-weight: 700;
  font-size: 1.4rem;
  text-transform: uppercase;
  height: 3.6rem;
  cursor: pointer;
  border-radius: 6px;
  transition: all 0.4s ease;
  outline: 0;

  &:hover {
    background: rgba(40, 40, 40, 0.1);
  }

  &:active {
    background: rgba(74, 108, 187, 0.2);
  }

  &:focus {
    border: 1px solid rgba(74, 108, 187, 0.6);
  }

  &:disabled {
    color: rgba(40, 40, 40, 0.3);
    background: rgba(40, 40, 40, 0.05);
    cursor: default;
  }
`;

export default Button;
