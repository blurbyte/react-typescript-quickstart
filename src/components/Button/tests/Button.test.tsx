import * as React from 'react';
import { shallow } from 'enzyme';

import Button from '../index';

describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
});
