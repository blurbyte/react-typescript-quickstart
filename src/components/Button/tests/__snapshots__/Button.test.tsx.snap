// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  background: rgba(40,40,40,0.05);
  border: none;
  color: #4a6cbb;
  border: 1px solid transparent;
  font-weight: 700;
  font-size: 1.4rem;
  text-transform: uppercase;
  width: 19.4rem;
  height: 3.6rem;
  cursor: pointer;
  border-radius: 6px;
  -webkit-transition: all 0.4s ease;
  transition: all 0.4s ease;
  outline: 0;
}

.c0:hover {
  background: rgba(40,40,40,0.1);
}

.c0:active {
  background: rgba(74,108,187,0.2);
}

.c0:focus {
  border: 1px solid rgba(74,108,187,0.6);
}

.c0:disabled {
  color: rgba(40,40,40,0.3);
  background: rgba(40,40,40,0.05);
  cursor: default;
}

<button
  className="c0"
/>
`;
